runtime/syntax/python.vim

```diff
@@ -84,13 +84,20 @@ syn keyword pythonStatement	as assert break continue del global
 syn keyword pythonStatement	lambda nonlocal pass return with yield
 syn keyword pythonStatement	class def nextgroup=pythonFunction skipwhite
 syn keyword pythonConditional	elif else if
-syn keyword pythonConditional	case match
 syn keyword pythonRepeat	for while
 syn keyword pythonOperator	and in is not or
 syn keyword pythonException	except finally raise try
 syn keyword pythonInclude	from import
 syn keyword pythonAsync		async await
 
+" Soft keywords
+" These keywords do not mean anything unless used in the right context
+" See https://peps.python.org/pep-0634/#the-match-statement for more on this.
+" In Python 3.10's IDLE for example, 'match' and 'case' are only highlighted
+" when in the right context
+syn match   pythonConditional   "\v^\s*case(.*:\s*$)@="
+syn match   pythonConditional   "\v^\s*match(.*:\s*$)@="
```

Comment:
A few issues with these regular expressions:

* Both `case` and `match` must be followed by a `\s` before any other content -- match subject or case pattern(s). These regular expressions are not enforcing this space separation and would accept `matchX` or `caseY`. There needs to be a `\s` as the first part of the content inside the parentheses for zero-width look-ahead match (`@=`).
* According to grammar, the `match subject:` must be followed by a newline, indented block, and dedent, which allows for the check of `$` at the end of the regex. The `case pattern:` is followed by a _block_ according to grammar. This means that besides "newline, indented block, dedent" it can be followed by a simple statement. So a check for `$` is not necessarily correct in that case. It can be a `\s*$`, but it could also be other characters that could make a simple statement between the colon and the end of the line.
* The patterns in this file _never_ use `\v` (very magic) and I'd keep the prevailing style. The non-capturing parentheses could be used `\%( ... \)@=` and are the only part of the regex that requires a slightly more complex writing without `\v` which doesn't seem too bad.
* The `^\s*` makes the space in front of `match` or `case` the part of this match, while only the keywords should be matched. See some examples in the other regular expressions in this file that set `\zs` for the start of the match after finding the pre-requisite part in front of the match.